src/dfi/consensus/governance.cpp

```diff
@@ -106,7 +101,39 @@ Res CGovernanceConsensus::operator()(const CGovernanceMessage &obj) const {
     return Res::Ok();
 }
 
-Res CGovernanceConsensus::operator()(const CGovernanceUnsetMessage &obj) const {
+Res CGovernanceConsensus::operator()(const CGovernanceClearHeightMessage &obj) const {
+    // Check auth
+    auto authCheck = AuthManager(blockCtx, txCtx);
+    if (auto res = authCheck.HasGovOrFoundationAuth(); !res) {
+        return res;
+    }
+
+    auto &mnview = blockCtx.GetView();
+
+    auto heightToClear = [](const auto &collection) {
+        std::set<uint64_t> heights;
+        for (const auto &[_, values] : collection) {
+            for (const auto &[height, _] : values) {
+                heights.insert(height);
+            }
+        }
+        return heights;
+    };
+
+    const auto setVars = mnview.GetAllStoredVariables();
+    for (const auto &height : heightToClear(setVars)) {
+        mnview.EraseStoredVariables(height);
+    }
+
+    const auto unsetVars = mnview.GetAllUnsetStoredVariables();
+    for (const auto &height : heightToClear(unsetVars)) {
+        mnview.EraseUnsetStoredVariables(height);
+    }
+
+    return Res::Ok();
+}
+
+Res CGovernanceConsensus::operator()(const CGovernanceUnsetHeightMessage &obj) const {
```

Comment:
should this have a DF24 check?

Reply:
https://github.com/DeFiCh/ain/pull/3063/files#diff-2e2980eda773148aea9f732c9f165bb7a9427b4f9df3c49b78b16d9166d58c42R289-R293

Reply:
ok thx. means the gates in other similar methods are actually unnecessary?

Reply:
yes, if it's only only used for simple gating tied to network upgrades.